app/javascript/react/screens/App/Overview/components/Migrations/MigrationsInProgressCard.js

```diff
@@ -63,15 +60,88 @@ const MigrationsInProgressCard = ({ plan, allRequestsWithTasks, reloadCard, hand
     tasks[task.source_id] = tasks[task.source_id] || {};
     tasks[task.source_id].completed = task.status === 'Ok' && task.state === 'finished';
     tasks[task.source_id].virtv2v_disks = task.options.virtv2v_disks;
+    tasks[task.source_id].playbooks = task.options.playbooks;
   });
 
   let completedVMs = 0;
   const totalVMs = Object.keys(tasks).length;
+  let taskRunningPreMigrationPlaybook = null;
+  let taskRunningPostMigrationPlaybook = null;
   Object.keys(tasks).forEach(task => {
     if (tasks[task].completed) completedVMs += 1;
+
+    // after we have assigned the latest tasks for each source, check the running playbook status
+    const { playbooks } = tasks[task];
+    if (playbooks) {
+      if (
+        playbooks.pre &&
+        (playbooks.pre.job_state === PLAYBOOK_JOB_STATES.PENDING ||
+          playbooks.pre.job_state === PLAYBOOK_JOB_STATES.ACTIVE)
+      ) {
+        taskRunningPreMigrationPlaybook = tasks[task];
+      }
+      if (
+        playbooks.post &&
+        (playbooks.post.job_state === PLAYBOOK_JOB_STATES.PENDING ||
+          playbooks.post.job_state === PLAYBOOK_JOB_STATES.ACTIVE)
+      ) {
+        taskRunningPostMigrationPlaybook = tasks[task];
+      }
+    }
   });
 
-  // UX business rule 3: reflect the total disk space migrated, aggregated across requests
+  let failedOverlay = null;
+  if (failed) {
+    failedOverlay = (
+      <OverlayTrigger
+        overlay={
+          <Popover id={`description_${plan.id}`} title={sprintf('%s', plan.name)}>
+            <Icon
+              type="pf"
+              name="error-circle-o"
+              size="sm"
+              style={{
+                width: 'inherit',
+                backgroundColor: 'transparent',
+                paddingRight: 5
+              }}
+            />
+            {sprintf(__('%s of %s VM migrations failed.'), failedVms, totalVMs)}
+          </Popover>
+        }
+        placement="top"
+        trigger={['hover']}
+        delay={500}
+        rootClose={false}
+      >
+        <Icon
+          type="pf"
+          name="error-circle-o"
+          size="md"
+          style={{
+            width: 'inherit',
+            backgroundColor: 'transparent',
+            paddingRight: 10
+          }}
+        />
+      </OverlayTrigger>
+    );
+  }
+
+  // UX business rule: if there are any pre migration playbooks running, show this content instead
+  if (taskRunningPreMigrationPlaybook) {
+    const playbookName = getPlaybookName(
+      serviceTemplatePlaybooks,
+      taskRunningPreMigrationPlaybook.playbooks.pre.job_id
```

Comment:
`job_id` is the ID for the associated `OrchestrationStack`

We can access the playbook ID from

**ServiceTemplateTransformationPlan**
```
{
    options:  {
        config_info:  {
            post_service_id: <id>,
            pre_service_id: <id>
        }
    }
}
```

Reply:
updated